### PR TITLE
LPS-144846 Create Modal to Add Object View

### PR DESCRIPTION
**Pull request by danvb112**
Related Issue: https://issues.liferay.com/browse/LPS-144846

### CODE REVIEW

**Comment by danvb112** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalBasicWithFieldName.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+			const {
+				title = Liferay.Language.get('an-error-occurred'),
+			} = await response.json();
+
+			setError(title);
+		}
+	};
+
+	const validate = ({name}: TInitialValues) => {
+		const errors: any = {};
```
Comment (miltonmc) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalBasicWithFieldName.tsx:91]: The more strict the better!

avoid use `any` type:
```suggestion
		const errors: {name?: string} = {};
```

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalBasicWithFieldName.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+		onSubmit,
+		validate,
+	});
+
+	return (
+		<>
+			<ClayModal observer={observer}>
+				<ClayForm onSubmit={handleSubmit}>
+					<ClayModal.Header>
+						{label === 'New Layout'
```
Comment (miltonmc) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalBasicWithFieldName.tsx:111]: Can this label be changed due to internationalization?

```suggestion
						{label === 'Liferay.Language.get('new-layout')'
```

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalBasicWithFieldName.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+							value={values.name[defaultLanguageId]}
+						/>
+					</ClayModal.Body>
+
+					<ClayModal.Footer
+						last={
+							<ClayButton.Group key={1} spaced>
+								<ClayButton
+									displayType="secondary"
+									onClick={() => onClose()}
```
Comment (miltonmc) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalBasicWithFieldName.tsx:137]: It seems you can directly call the `onClose`:
```suggestion
									onClick={onClose}
```

**Review by miltonmc — CHANGES_REQUESTED** [review]
I've found some minor issues:

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:forward

**Pull request closed without merging**